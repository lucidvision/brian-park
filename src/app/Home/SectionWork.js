import React from 'react'
import styled from 'styled-components'
import { HeadingSecondary, ImageCoastline, Paragraph } from 'components'
import { media, utilities } from 'styles'

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: 1rem;

  ${media.phone`
    margin-bottom: ${utilities.marginMedium};
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;
  margin-right: -4rem;

  ${media.phone`
    flex-direction: column;
    align-items: center;
    margin-right: 0;
  `}
`

const StyledDivInfo = styled.div`
  margin-top: 8rem;

  ${media.tabLand`
    margin-top: 6rem;
  `}

  ${media.tabPort`
    margin-top: 4rem;
  `}

  ${media.phone`
    margin-top: 0;
    text-align: center;
  `}
`

const StyledImageCoastline = styled(ImageCoastline)`
  height: 50%;
  width: 50%

  ${media.phone`
    height: 100%;
    width: 100%;
  `}
`

export default function SectionWork (props) {
  return (
    <section>
      <StyledHeadingSecondary text='Coastline Market' />
      <StyledDiv>
        <StyledDivInfo>
          <Paragraph text={'As CTO, I led the development of Coastline\'s "Harvest-on-Demand" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API.'} /><br />
          <Paragraph text={'Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition.'} /><br />
          <Paragraph text={'Through an iterative development process, Coastline delights customers with accurate design and product-market fit.'} /><br />
        </StyledDivInfo>
        <StyledImageCoastline />
      </StyledDiv>
    </section>
  )
}
