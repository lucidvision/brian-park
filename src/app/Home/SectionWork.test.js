import React from 'react'
import { shallow } from 'enzyme'
import SectionWork from './SectionWork'

describe('SectionWork', () => {
  const sectionWork = shallow(<SectionWork />)

  it('renders correctly', () => {
    expect(sectionWork).toMatchSnapshot()
  })

  it('renders a secondary heading', () => {
    expect(sectionWork.find('Styled(HeadingSecondary)').props().text).toBe('Coastline Market')
  })

  describe('Paragraphs', () => {
    it('renders 3 paragraphs', () => {
      expect(sectionWork.find('Paragraph')).toHaveLength(3)
    })

    it('renders first paragraph', () => {
      expect(sectionWork.find('Paragraph').at(0).props().text).toBe('As CTO, I led the development of Coastline\'s "Harvest-on-Demand" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API.')
    })

    it('renders second paragraph', () => {
      expect(sectionWork.find('Paragraph').at(1).props().text).toBe('Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition.')
    })

    it('renders third paragraph', () => {
      expect(sectionWork.find('Paragraph').at(2).props().text).toBe('Through an iterative development process, Coastline delights customers with accurate design and product-market fit.')
    })
  })

  it('renders the coastline app image', () => {
    expect(sectionWork.find('Styled(ImageCoastline)').exists()).toBeTruthy()
  })
})
