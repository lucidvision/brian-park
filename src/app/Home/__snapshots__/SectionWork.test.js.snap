// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionWork renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionWork />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(HeadingSecondary)
          text="Coastline Market"
/>,
        <styled.div>
          <styled.div
                    className="wow slideInLeft"
          >
                    <Paragraph
                              color="rgba(0, 13, 85, 0.8)"
                              size="1.8rem"
                              text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
                    />
                    <br />
                    <Paragraph
                              color="rgba(0, 13, 85, 0.8)"
                              size="1.8rem"
                              text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
                    />
                    <br />
                    <Paragraph
                              color="rgba(0, 13, 85, 0.8)"
                              size="1.8rem"
                              text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
                    />
                    <br />
          </styled.div>
          <Styled(ImageCoastline)
                    className="wow slideInRight"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "text": "Coastline Market",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              className="wow slideInLeft"
>
              <Paragraph
                            color="rgba(0, 13, 85, 0.8)"
                            size="1.8rem"
                            text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
              />
              <br />
              <Paragraph
                            color="rgba(0, 13, 85, 0.8)"
                            size="1.8rem"
                            text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
              />
              <br />
              <Paragraph
                            color="rgba(0, 13, 85, 0.8)"
                            size="1.8rem"
                            text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
              />
              <br />
</styled.div>,
            <Styled(ImageCoastline)
              className="wow slideInRight"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Paragraph
                  color="rgba(0, 13, 85, 0.8)"
                  size="1.8rem"
                  text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
/>,
                <br />,
                <Paragraph
                  color="rgba(0, 13, 85, 0.8)"
                  size="1.8rem"
                  text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
/>,
                <br />,
                <Paragraph
                  color="rgba(0, 13, 85, 0.8)"
                  size="1.8rem"
                  text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
/>,
                <br />,
              ],
              "className": "wow slideInLeft",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "rgba(0, 13, 85, 0.8)",
                  "size": "1.8rem",
                  "text": "As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "rgba(0, 13, 85, 0.8)",
                  "size": "1.8rem",
                  "text": "Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "rgba(0, 13, 85, 0.8)",
                  "size": "1.8rem",
                  "text": "Through an iterative development process, Coastline delights customers with accurate design and product-market fit.",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "wow slideInRight",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(HeadingSecondary)
            text="Coastline Market"
/>,
          <styled.div>
            <styled.div
                        className="wow slideInLeft"
            >
                        <Paragraph
                                    color="rgba(0, 13, 85, 0.8)"
                                    size="1.8rem"
                                    text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
                        />
                        <br />
                        <Paragraph
                                    color="rgba(0, 13, 85, 0.8)"
                                    size="1.8rem"
                                    text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
                        />
                        <br />
                        <Paragraph
                                    color="rgba(0, 13, 85, 0.8)"
                                    size="1.8rem"
                                    text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
                        />
                        <br />
            </styled.div>
            <Styled(ImageCoastline)
                        className="wow slideInRight"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "text": "Coastline Market",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                className="wow slideInLeft"
>
                <Paragraph
                                color="rgba(0, 13, 85, 0.8)"
                                size="1.8rem"
                                text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
                />
                <br />
                <Paragraph
                                color="rgba(0, 13, 85, 0.8)"
                                size="1.8rem"
                                text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
                />
                <br />
                <Paragraph
                                color="rgba(0, 13, 85, 0.8)"
                                size="1.8rem"
                                text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
                />
                <br />
</styled.div>,
              <Styled(ImageCoastline)
                className="wow slideInRight"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Paragraph
                    color="rgba(0, 13, 85, 0.8)"
                    size="1.8rem"
                    text="As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API."
/>,
                  <br />,
                  <Paragraph
                    color="rgba(0, 13, 85, 0.8)"
                    size="1.8rem"
                    text="Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition."
/>,
                  <br />,
                  <Paragraph
                    color="rgba(0, 13, 85, 0.8)"
                    size="1.8rem"
                    text="Through an iterative development process, Coastline delights customers with accurate design and product-market fit."
/>,
                  <br />,
                ],
                "className": "wow slideInLeft",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "rgba(0, 13, 85, 0.8)",
                    "size": "1.8rem",
                    "text": "As CTO, I led the development of Coastline's \\"Harvest-on-Demand\\" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "rgba(0, 13, 85, 0.8)",
                    "size": "1.8rem",
                    "text": "Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "rgba(0, 13, 85, 0.8)",
                    "size": "1.8rem",
                    "text": "Through an iterative development process, Coastline delights customers with accurate design and product-market fit.",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "wow slideInRight",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
